### PR TITLE
Split discrete trajectory

### PR DESCRIPTION
**Pull request by jnetosoares**
## Types of changes

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [ ] I have read the **CONTRIBUTING** document.
-   [ ] My code follows the code style of this project (see `.code-style.md`).
-   [ ] All new and existing tests passed (see `.testing.md`).
-   [ ] I have added tests to cover my changes.
-   [ ] My change requires a change to the documentation.
-   [ ] I have updated the documentation accordingly (see `.documentation.md`).

## Description

-   What is the current behavior? (You can also link to an open issue here)
    

-   What is the new behavior (if this is a feature change)?

### CODE REVIEW

File: pymove/core/pandas_discrete.py [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+from pymove.core.dataframe import MoveDataFrame
+from pymove.core.grid import Grid
+from pymove.core.pandas import PandasMoveDataFrame
+from pymove.preprocessing.filters import clean_trajectories_with_few_points
+from pymove.preprocessing.segmentation import (
+    _drop_single_point,
+    _prepare_segmentation,
+    _update_curr_tid_count,
+)
+from pymove.utils.constants import (
```
Comment (regispires) [inline pymove/core/pandas_discrete.py:13]: Codacy found an issue: ['pymove.utils.constants.GRID_ID' imported but unused (F401)](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=6293208)

File: pymove/core/pandas_discrete.py [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+
+                filter_ = []
+                for index, row in md.iterrows():
+                    local_label_ = row[local_label]
+                    prev_local = row[PREV_LOCAL]
+                    threshold = statistics[
+                        (statistics[local_label]
+                         == local_label_) & (statistics[PREV_LOCAL] == prev_local)
+                    ][THRESHOLD].values
+                    time_row = row[TIME_TO_PREV]
```
Comment (regispires) [inline pymove/core/pandas_discrete.py:292]: Codacy found an issue: [local variable 'time_row' is assigned to but never used (F841)](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=6293208)

File: pymove/core/pandas_discrete.py (hunk 1 above)
Comment (regispires) [inline pymove/core/pandas_discrete.py:13]: Codacy found an issue: [Unused GRID_ID imported from pymove.utils.constants](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=6293208)

File: pymove/core/pandas_discrete.py (hunk 2 above)
Comment (regispires) [inline pymove/core/pandas_discrete.py:292]: Codacy found an issue: [Unused variable 'time_row'](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=6293208)

File: pymove/__init__.py [hunk 3]
```diff
@@ -12,6 +12,7 @@
 from .core.dask import DaskMoveDataFrame
 from .core.dataframe import MoveDataFrame
 from .core.pandas import PandasMoveDataFrame
+from .core.pandas_discrete import PandasDiscreteMoveDataFrame
```
Comment (regispires) [inline pymove/__init__.py:15]: Codacy found an issue: ['.core.pandas_discrete.PandasDiscreteMoveDataFrame' imported but unused (F401)](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=6293208)

File: pymove/core/pandas_discrete.py [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+                                               min_points_per_trajectory=2, inplace=True)
+
+            current_tid, ids, count = _prepare_segmentation(data_, label_id, TID_STAT)
+
+            for idx in progress_bar(ids, desc='Generating %s' % TID_STAT):
+                md = data_.loc[idx, [TIME_TO_PREV, local_label, PREV_LOCAL]]
+                md = pd.DataFrame(md)
+
+                filter_ = []
+                for index, row in md.iterrows():
```
Comment (regispires) [inline pymove/core/pandas_discrete.py:285]: Codacy found an issue: [Unused variable 'index'](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=6293208)

File: pymove/core/pandas_discrete.py [hunk 5]
```diff
@@ -0,0 +1,315 @@
...
+    def __init__(
+        self,
+        data,
+        latitude=LATITUDE,
+        longitude=LONGITUDE,
+        datetime=DATETIME,
+        traj_id=TRAJ_ID,
+        local_label=LOCAL_LABEL
+    ):
+        """
```
Comment (regispires) [inline pymove/core/pandas_discrete.py:45]: Codacy found an issue: [No blank lines allowed after function docstring (found 1) (D202)](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=6293208)

File: pymove/core/pandas_discrete.py [hunk 6]
```diff
@@ -43,6 +42,7 @@ def __init__(
         local_label=LOCAL_LABEL
     ):
         """
+
         Checks whether past data has 'lat', 'lon', 'datetime' and 'local_label'
```
Comment (regispires) [inline pymove/core/pandas_discrete.py:46]: Codacy found an issue: ['pymove.utils.constants.GRID_ID' imported but unused (F401)](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=6293208)

Comment (regispires) [inline pymove/core/pandas_discrete.py:46]: Codacy found an issue: [Unused GRID_ID imported from pymove.utils.constants](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=6293208)

**Review by flych3r — APPROVED** [review]

**Pull request merged**